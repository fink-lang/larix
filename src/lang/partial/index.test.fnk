{decribe, it, expect, to_match_snapshot, to_throw} = import '../../testing/jest'

{parse_expr} = import '../../'


describe:: 'partial', fn:

  it:: 'parses call', fn:
    expect::
      parse_expr:: `foo:: ?`
      to_match_snapshot

    expect::
      parse_expr:: `foo.bar:: ?`
      to_match_snapshot


  it:: 'parses member', fn:
    expect::
      parse_expr:: `?.bar()`
      to_match_snapshot


  it:: 'parses binary', fn:
    expect::
      parse_expr:: `? > 1`
      to_match_snapshot
    expect::
      parse_expr:: `1 < ?`
      to_match_snapshot
    expect::
      parse_expr:: `1 + ? * 3`
      to_match_snapshot


  it:: 'parses str', fn:
    expect::
      parse_expr:: `'foo \${?} bar'`
      to_match_snapshot

