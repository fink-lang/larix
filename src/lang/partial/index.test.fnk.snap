// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`partial parses binary 1`] = `
comp > (1:0-1:5)
  partial (1:0-1:1) ?
  number (1:4-1:5) 1
`;

exports[`partial parses binary 2`] = `
comp < (1:0-1:5)
  number (1:0-1:1) 1
  partial (1:4-1:5) ?
`;

exports[`partial parses binary 3`] = `
arithm + (1:0-1:9)
  number (1:0-1:1) 1
  arithm * (1:4-1:9)
    partial (1:4-1:5) ?
    number (1:8-1:9) 3
`;

exports[`partial parses call 1`] = `
call (1:0-1:6)
  ident (1:0-1:3) foo
  partial (1:6-1:7) ?

`;

exports[`partial parses call 2`] = `
call (1:0-1:10)
  member . (1:0-1:7)
    ident (1:0-1:3) foo
    ident (1:4-1:7) bar
  partial (1:10-1:11) ?

`;

exports[`partial parses member 1`] = `
call (1:0-1:7)
  member . (1:0-1:5)
    partial (1:0-1:1) ?
    ident (1:2-1:5) bar

`;

exports[`partial parses str 1`] = `
string ' (1:0-1:14)
  \`foo \`
    partial (1:7-1:8) ?
  \` bar\`
`;
